spec/spec_helper.rb

```diff
@@ -0,0 +1,91 @@
+# a separate helper file that requires the additional dependencies and performs
+# the additional setup, and require it from the spec files that actually need
+# it.
+#
+# The `.rspec` file also contains a few flags that are not defaults but that
+# users commonly want.
+#
+# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+RSpec.configure do |config|
+  # rspec-expectations config goes here. You can use an alternate
+  # assertion/expectation library such as wrong or the stdlib/minitest
+  # assertions if you prefer.
+  config.expect_with :rspec do |expectations|
+    # This option will default to `true` in RSpec 4. It makes the `description`
+    # and `failure_message` of custom matchers include text for helper methods
+    # defined using `chain`, e.g.:
+    #     be_bigger_than(2).and_smaller_than(4).description
+    #     # => "be bigger than 2 and smaller than 4"
+    # ...rather than:
+    #     # => "be bigger than 2"
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  # rspec-mocks config goes here. You can use an alternate test double
+  # library (such as bogus or mocha) by changing the `mock_with` option here.
+  config.mock_with :rspec do |mocks|
+    # Prevents you from mocking or stubbing a method that does not exist on
+    # a real object. This is generally recommended, and will default to
+    # `true` in RSpec 4.
+    mocks.verify_partial_doubles = true
+  end
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+
+# These two settings work together to allow you to limit a spec run
+# to individual examples or groups you care about by tagging them with
+# `:focus` metadata. When nothing is tagged with `:focus`, all examples
+# get run.
+config.filter_run :focus
+config.run_all_when_everything_filtered = true
+
+# Allows RSpec to persist some state between runs in order to support
+# the `--only-failures` and `--next-failure` CLI options. We recommend
+# you configure your source control system to ignore this file.
+config.example_status_persistence_file_path = "spec/examples.txt"
+
+# Limits the available syntax to the non-monkey patched syntax that is
+# recommended. For more details, see:
+#   - http://rspec.info/blog/2012/06/rspecs-new-expectation-syntax/
+#   - http://www.teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/
+#   - http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#zero-monkey-patching-mode
+config.disable_monkey_patching!
+
+# Many RSpec users commonly either run the entire suite or an individual
+# file, and it's useful to allow more verbose output when running an
+# individual spec file.
+if config.files_to_run.one?
+  # Use the documentation formatter for detailed output,
+  # unless a formatter has already been configured
+  # (e.g. via a command-line flag).
+  config.default_formatter = 'doc'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.